File: queenslize/src/main/java/be/ordina/hp/pact/queenslize/domain/Pizza.java
package be.ordina.hp.pact.queenslize.domain;

import javax.persistence.Entity;
import javax.persistence.Id;
import java.math.BigDecimal;
import java.util.UUID;

@Entity
public class Pizza {

    @Id
    private UUID id;

    private String name;

    private String price;

    private boolean vegetarian;

    private BigDecimal diameter;

    Pizza() {
        //empty JPA/JSON constructor
    }

    public Pizza(String name, String price, boolean vegetarian, BigDecimal diameter) {

        this.id = UUID.randomUUID();
        this.name = name;
        this.price = price.toString();
        this.vegetarian = vegetarian;
        this.diameter = diameter;
    }

    public UUID getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getPrice() {
        return price;
    }

    public boolean isVegetarian() {
        return vegetarian;
    }

    public BigDecimal getDiameter() {
        return diameter;
    }
}

File: queenslize/src/main/java/be/ordina/hp/pact/queenslize/domain/Pizza.kt
package be.ordina.hp.pact.queenslize.domain

import java.math.BigDecimal
import java.util.*
import javax.persistence.Entity
import javax.persistence.Id

@Entity
data class Pizza(@Id val id: UUID = UUID.randomUUID(),
                 val name: String = "",
                 val price: BigDecimal = BigDecimal.ZERO,
                 val isVegetarian: Boolean = false,
                 val diameter: BigDecimal = BigDecimal.ZERO)
